lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -606,6 +611,15 @@ data AddressAmount addr = AddressAmount
     , amount :: !(Quantity "lovelace" Natural)
     } deriving (Eq, Generic, Show)
 
+newtype ApiAddressInspect = ApiAddressInspect
+    { unApiAddressInspect :: Aeson.Value }
+    deriving (Eq, Generic, Show)
+
+newtype ApiAddressInspectData = ApiAddressInspectData
+    { unApiAddressInspectData :: Text }
+    deriving (Eq, Generic, Show)
+    deriving newtype (IsString)
```

Comment:
could you comment why this was needed and used? (especially if we have Show above)

Reply:
It allows for using the `OverloadedString` extension in a transparent fashion for literal strings like we do for `Text` and `ByteString`. So, one can write: 

```hs
let inspectData = "patate"
```

instead of having to:

```hs
let inspectData = ApiAddressInspectData "patate"
```